src/main/java/org/broadinstitute/consent/http/service/SummaryService.java

```diff
@@ -510,11 +509,11 @@ private void setSummaryHeaderDataAccessRequest(FileWriter summaryWriter , Intege
         );
     }
 
-    private Document findAssociatedDAR(List<Document> dataAccessRequests, String referenceId) {
-        Optional<Document> documentOption = dataAccessRequests.stream().
-                filter(d -> d.getString(DarConstants.REFERENCE_ID).equalsIgnoreCase(referenceId)).
+    private DataAccessRequest findAssociatedDAR(List<DataAccessRequest> dataAccessRequests, String referenceId) {
+        Optional<DataAccessRequest> darOption = dataAccessRequests.stream().
+                filter(d -> Objects.nonNull(d) ? d.getReferenceId().equalsIgnoreCase(referenceId) : false).
                 findFirst();
-        return documentOption.orElse(null);
+        return darOption.orElse(null);
     }
```

Comment:
This method is another opportunity for refactoring. Instead of calling this method, the caller should instead create a `Map<String, DataAccessRequest>` and just access the map by the reference id. Not for this PR unless you're up for it.